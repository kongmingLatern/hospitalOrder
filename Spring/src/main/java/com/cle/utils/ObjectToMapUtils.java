package com.cle.utils;

import java.lang.reflect.Field;
import java.util.HashMap;
import java.util.Map;

public class ObjectToMapUtils {

    public static Map<String, Object> convertObjToMap(Object obj) throws Exception{

        Map<String, Object> map = new HashMap<String, Object>();

        Field[] fields = obj.getClass().getDeclaredFields();

        for(int i = 0; i < fields.length; i++){

            Field field = obj.getClass().getDeclaredField(fields[i].getName());
            field.setAccessible(true);
            Object value = field.get(obj);

            map.put(fields[i].getName(), value);
        }

        return map;
    }

}