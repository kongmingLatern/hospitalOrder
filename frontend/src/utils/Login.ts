export default class LoginState {
  #uid: string = ''
  #username: string = ''
  #isAuth: boolean = false

  constructor(uid: string, username: string, isAuth: boolean) {
    this.#uid = uid
    this.#username = username
    this.#isAuth = isAuth
  }

  get uid() {
    return this.#uid
  }

  get username() {
    return this.#username
  }

  get isAuth() {
    return this.#isAuth
  }


  set uid(uid: string) {
    this.#uid = uid
  }

  set username(username: string) {
    this.#username = username
  }

  set isAuth(isAuth: boolean) {
    this.#isAuth = isAuth
  }


  setLocalStorage() {
    localStorage.setItem('uid', this.#uid)
    localStorage.setItem('username', this.#username)
    localStorage.setItem('isAuth', this.#isAuth ? '1' : '0')
  }

  getLocalStorage(key: string): string | null {
    return localStorage.getItem(key)
  }

}