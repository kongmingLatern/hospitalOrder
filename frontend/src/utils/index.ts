export function randomString() {
  let len = 32;
  let chars = 'ABCDEFGHJKMNPQRSTWXYZabcdefhijkmnprstwxyz2345678';
  let maxPos = chars.length;
  let character = '';
  for (let i = 0; i < len; i++) {
    character += chars.charAt(Math.floor(Math.random() * maxPos))
  }
  return character;
}

export function formatData(data: Record<string, any>[]) {
  return data.map(item => {
    return {
      ...item,
      isCancel: item.isCancel ? '是' : '否',
      isFinish: item.isFinish ? '是' : '否',
    }
  })
};

export function formatObject(data: Record<string, any>) {
  for (const key in data) {
    if (key === 'cancelCount') {
      continue;
    }
    if (data[key] === 0) {
      data[key] = '否'
    } else if (data[key] === 1) {
      data[key] = '是'
    }
  }
  return data
};
export function hasOwnProperty<T, K extends PropertyKey>(
  obj: T,
  prop: K
): obj is T & Record<K, unknown> {
  return Object.prototype.hasOwnProperty.call(obj, prop);
}