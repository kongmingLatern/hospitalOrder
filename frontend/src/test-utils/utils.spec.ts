import { describe, it, expect } from 'vitest'
import { formatData, formatObject, getStartEndTime, randomString, formatDate } from '../utils';
describe('randomString', () => {
  it.skip('Random string length = 32', () => {
    expect(randomString()).toMatchInlineSnapshot('"TBfsJX2MJimRsKpN3hKjJek4eCSf2Y5E"')
  })
})

describe('format data', () => {
  it('format dataSource', () => {
    const dataSource = [
      {
        isCancel: 0,
        isFinish: 0,
      }, {
        isCancel: 1,
        isFinish: 1,
      }]
    expect(formatData(dataSource)).toEqual([
      {
        isCancel: '否',
        isFinish: '否',
      }, {
        isCancel: '是',
        isFinish: '是',
      }]
    )
  });

  it('format object1', () => {
    const dataSource =
    {
      isCancel: 0,
      isFinish: 0,
    }
    expect(formatObject(dataSource)).toEqual(
      {
        isCancel: '否',
        isFinish: '否',
      })
  });
  it('format object2', () => {
    const dataSource =
    {
      isAuth: 0,
      isAllow: 0,
    }
    expect(formatObject(dataSource)).toEqual(
      {
        isAuth: '否',
        isAllow: '否',
      })
  });
  it('format object3', () => {
    const dataSource =
    {
      cancelCount: 0,
      isAuth: 0,
      isAllow: 0,
    }
    expect(formatObject(dataSource)).toEqual(
      {
        cancelCount: 0,
        isAuth: '否',
        isAllow: '否',
      })
  });
})

describe('test date', () => {
  it.skip('should get tomorrow date', () => {
    const result = getStartEndTime(-1)
    expect(result[0]).toMatchInlineSnapshot('1664899200')
  })

  it.skip('should get current date', () => {
    const selectId = 1
    let time: number
    if (selectId === 1) {
      time = getStartEndTime(-1)[0] + 8 * 60 * 60 * 1000
    } else {
      time = getStartEndTime(-1)[0] + 13 * 60 * 60 * 1000
    }
    expect(time).toMatchInlineSnapshot('"2023-09-03 08:00:00"')
    expect(formatDate(time)).toMatchInlineSnapshot('')
  })
});