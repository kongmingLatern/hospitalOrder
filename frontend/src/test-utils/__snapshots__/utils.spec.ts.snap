// Vitest Snapshot v1

exports[`randomString > Random string length = 32 1`] = `"4ykfsP8SNGKMfZGyRmkXF8yWTNCirjiA"`;
